### PR TITLE
Scorecards parser

### PR DESCRIPTION
**Pull request by lumjjb**
Depends on #127

### CODE REVIEW

**Comment by lumjjb** [comment]
ready for review!

File: pkg/assembler/nodes.go [hunk 1]
```diff
@@ -314,3 +350,46 @@ func (e ContainsEdge) PropertyNames() []string {
...
+}
+
+func (e MetadataForEdge) Type() string {
+	return "MetadataFor"
+}
+
+func (e MetadataForEdge) Nodes() (v, u GuacNode) {
+	uA, uP := isDefined(e.ForArtifact), isDefined(e.ForPackage)
+	if uA == uP {
+		panic("only on of package and artifact dependency node defined for DependsOn relationship")
```
Comment (mlieberman85) [inline pkg/assembler/nodes.go:372]: nit: "one" is spelled wrong

File: pkg/ingestor/parser/scorecard/parser_scorecard.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	mnNode := assembler.MetadataNode{
+		MetadataType: "scorecard",
+		ID:           metadataId(s),
+		Details:      map[string]interface{}{},
+	}
+
+	for _, c := range s.Checks {
+		mnNode.Details[strings.ReplaceAll(c.Name, "-", "_")] = c.Score
+	}
+	mnNode.Details["repo"] = s.Repo.Name
```
Comment (pxp928) [inline pkg/ingestor/parser/scorecard/parser_scorecard.go:110]: would be better to make the keys of the map constants defied in assembler.

**Comment by lumjjb** [comment]
addressed comments! @pxp928 @mlieberman85 ready for review!

**Review by pxp928 — APPROVED** [review]
LGTM

File: pkg/assembler/nodes.go [hunk 3]
```diff
@@ -318,3 +354,46 @@ func (e ContainsEdge) PropertyNames() []string {
...
+
+// MetadataFor is an edge that represents the fact that an
+// a metadata node represents metadata for an `ArtifactNode/PackageNode`
+// Only one of each side of the edge should be defined.
+type MetadataForEdge struct {
+	// From node
+	MetadataNode MetadataNode
+	// To node
+	ForArtifact ArtifactNode
+	ForPackage  PackageNode
```
Comment (mihaimaruseac) [inline pkg/assembler/nodes.go:366]: I'm not sure we can have an edge with 3 nodes. Looking at Nodes() below it looks like we either have an artifact or a package, but never both? Should we document this? (more of a nit)

File: pkg/ingestor/parser/scorecard/parser_scorecard.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		return fmt.Errorf("expected document type: %v, actual document type: %v", processor.DocumentScorecard, doc.Type)
+	}
+
+	switch doc.Format {
+	case processor.FormatJSON:
+		var scorecard sc.JSONScorecardResultV2
+		if err := json.Unmarshal(doc.Blob, &scorecard); err != nil {
+			return err
+		}
+		//p.scorecards = append(s.scorecards, scorecard)
```
Comment (mihaimaruseac) [inline pkg/ingestor/parser/scorecard/parser_scorecard.go:58]: Nit: I think we can remove the comment

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/assembler/nodes.go (hunk 3 above)
Comment (lumjjb) [reply pkg/assembler/nodes.go:366]: hmm - i added a comment in the struct definition - do you think we need to be more explicit?

Comment (mihaimaruseac) [reply pkg/assembler/nodes.go:366]: I think we should be good

**Pull request merged**